sdgym/datasets.py

```diff
@@ -312,7 +312,12 @@ def get_dataset_paths(
                 if _path_contains_data_and_metadata(dataset) and dataset not in datasets:
                     datasets.append(dataset)
         else:
-            datasets = _get_available_datasets(modality, bucket=bucket)
+            datasets = _get_available_datasets(
+                modality,
+                bucket=bucket,
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key
```

Comment:
Discovered this bug while bug hunting on collab. Locally was hard to spot since the default `s3` client would use my private credentials.